### ISSUE TITLE
[BUG] Kotlin version

### ISSUE THREAD

**Issue by Elabar** [Nozbe/WatermelonDB#1147]
It seems like this library is using some type that is only available in Kotlin 1.3.50 and up?
But the installation guide still suggest to use Kotlin 1.3.21

Error I got when I define the kotlinVersion in buildscript.ext as 1.3.21
```
e: E:\dev\ap\node_modules\@nozbe\watermelondb\native\android\src\main\java\com\nozbe\watermelondb\Database.kt: (37, 39): 'forEach((T) -> Unit): Unit' is only available since Kotlin 1.3.50 and cannot be used in Kotlin 1.3
e: E:\dev\ap\node_modules\@nozbe\watermelondb\native\android\src\main\java\com\nozbe\watermelondb\Database.kt: (56, 34): 'forEach((T) -> Unit): Unit' is only available since Kotlin 1.3.50 and cannot be used in Kotlin 1.3
```

Problem is solved by defining the kotlinVersion to 1.3.50 in android/build.gradle file like so
```
buildscript {
    ext {
        ....
        kotlinVersion = "1.3.50"
    }
}
```

**Comment by rozPierog** [comment]
@Elabar can you open a PR with fix to the docs? 
I'm gonna create a task for myself to make kotlin version less strict

**Comment by Elabar** [comment]
Sure, here you go.
https://github.com/Nozbe/WatermelonDB/pull/1152

**Comment by AntoineC-dev** [comment]
> It seems like this library is using some type that is only available in Kotlin 1.3.50 and up? But the installation guide still suggest to use Kotlin 1.3.21
> 
> Error I got when I define the kotlinVersion in buildscript.ext as 1.3.21
> 
> ```
> e: E:\dev\ap\node_modules\@nozbe\watermelondb\native\android\src\main\java\com\nozbe\watermelondb\Database.kt: (37, 39): 'forEach((T) -> Unit): Unit' is only available since Kotlin 1.3.50 and cannot be used in Kotlin 1.3
> e: E:\dev\ap\node_modules\@nozbe\watermelondb\native\android\src\main\java\com\nozbe\watermelondb\Database.kt: (56, 34): 'forEach((T) -> Unit): Unit' is only available since Kotlin 1.3.50 and cannot be used in Kotlin 1.3
> ```
> 
> Problem is solved by defining the kotlinVersion to 1.3.50 in android/build.gradle file like so
> 
> ```
> buildscript {
>     ext {
>         ....
>         kotlinVersion = "1.3.50"
>     }
> }
> ```

It worked ! Thank you

**Comment by Elabar** [comment]
Just FYI, as far as the warning you reply here, should be fine (my experience so far)
Yes, RNScreen v3.8 sort of break the compatibility with Kotlin v1.3.X. This issue is acknowledged by them and it is fixed (probably release in next version)

https://github.com/software-mansion/react-native-screens/pull/1165

Moreover, maintainer said he is going to make the kotlin version less strict.

> @Elabar can you open a PR with fix to the docs? I'm gonna create a task for myself to make kotlin version less strict

**Comment by AgusMaris** [comment]
> 

I had to do the same and its working like a charm now, thanks!

**Comment by AgusMaris** [comment]
> Hi everyone. I think this should not be a closed issue since defining the kotlin version in buildscript leads to errors with other libraries such as react-native-screens (used for navigation)
> 
> -- Error :
> 
> ## `The Kotlin Gradle plugin was loaded multiple times in different subprojects, which is not supported and may break the build. This might happen in subprojects that apply the Kotlin plugins with the Gradle 'plugins { ... }' DSL if they specify explicit versions, even if the versions are equal. Please add the Kotlin plugin to the common parent project or the root project, then remove the versions in the subprojects. If the parent project does not need the plugin, add 'apply false' to the plugin line. See: https://docs.gradle.org/current/userguide/plugins.html#sec:subprojects_plugins_dsl The Kotlin plugin was loaded in the following projects: ':nozbe_watermelondb', ':react-native-screens'`
> Since it would break my build i had to change kotlin version from "1.3.50" (previous fix) to "1.4.10" which is the version used in react-native-screens library dependencies classpath.

I had to do the same and its working like a charm now, thanks!
